python/graphstorm/dataloading/dataset.py

```diff
@@ -66,8 +68,15 @@ def prepare_batch_input(g, input_nodes,
 
         if feat_name is not None:
             # concatenate multiple features together
-            feat[ntype] = th.cat([g.nodes[ntype].data[fname][nid].to(dev) \
-                for fname in feat_name], dim=1)
+            feats = []
+            for fname in feat_name:
+                data = g.nodes[ntype].data[fname]
+                if hasattr(data, 'gather') and is_wholegraph_embedding(data):
```

Comment:
Do we need to check ``hasattr(data, 'gather')``? I think we don't

Reply:
I think its safer to put a check. In future when we will add NVSHMEM it will have a different attribute.

Reply:
Removed.